spec/models/ucas_matched_application_spec.rb

```diff
@@ -1,77 +1,70 @@
 require 'rails_helper'
 
 RSpec.describe UCASMatchedApplication do
-  let(:course) { create(:course, recruitment_cycle_year: 2020) }
-  let(:candidate) { create(:candidate) }
-  let(:course_option) { create(:course_option, course: course) }
-  let(:application_choice) { create(:application_choice, course_option: course_option) }
-  let(:application_form) { create(:completed_application_form, candidate_id: candidate.id, application_choices: [application_choice]) }
-  let(:apply_again_application_form) { create(:application_form, candidate_id: candidate.id) }
   let(:recruitment_cycle_year) { 2020 }
-  let(:candidate1) { create(:candidate) }
-  let(:application_choice1) { create(:application_choice, :with_accepted_offer, course_option: course_option) }
-  let(:application_form1) { create(:completed_application_form, candidate_id: candidate1.id, application_choices: [application_choice1]) }
-  let(:candidate2) { create(:candidate) }
-  let(:application_choice2) { create(:application_choice, :with_rejection, course_option: course_option) }
-  let(:application_form2) { create(:completed_application_form, candidate_id: candidate2.id, application_choices: [application_choice2]) }
-
-  before do
-    apply_again_application_form
-    application_form
-    application_form1
-    application_form2
-    create(:course, code: course.code, provider: course.provider, recruitment_cycle_year: 2021)
-  end
+  let(:candidate) { build_stubbed(:candidate) }
+  let(:provider) { build_stubbed(:provider) }
+  let(:course) { build_stubbed(:course, recruitment_cycle_year: recruitment_cycle_year, provider: provider) }
+  let(:course_option) { build_stubbed(:course_option, course: course) }
 
   describe '#course' do
+    let!(:course) { create(:course, recruitment_cycle_year: 2020) }
+
     it 'returns the course for the correct recruitment cycle' do
       ucas_matching_data =
         { 'Course code' => course.code.to_s,
           'Provider code' => course.provider.code.to_s,
           'Apply candidate ID' => candidate.id.to_s }
-      ucas_matching_application = UCASMatchedApplication.new(ucas_matching_data, recruitment_cycle_year)
+      ucas_matching_application = UCASMatchedApplication.new(ucas_matching_data, course.recruitment_cycle_year)
 
       expect(ucas_matching_application.course).to eq(course)
     end
 
-    it 'returns the course details for a course which is not on Apply' do
-      ucas_matching_data =
-        { 'Scheme' => 'U',
+    context 'when a course is not on Apply' do
+      it 'returns the correct course details' do
+        allow(Provider).to receive(:find_by).with(code: course.provider.code.to_s).and_return(course.provider)
```

Comment:
Suggestion—as we're creating db records anyway, can remove this for clarity in exchange for a minor performance hit.